### PR TITLE
Avoid index_active_storage_attachments_uniqueness violation by attaching multiple files in a single call to attach

### PR DESCRIPTION
**Pull request by jpawlyn**
# Description
We recently tried to switch to using `direct_upload: true` on the `Files` field. However when creating a new record with multiple files, we got the error `duplicate key value violates unique constraint "index_active_storage_attachments_uniqueness"`.

The fix for this is to attach files in a single call to `attach` instead of calling `attach` for each file.

Fixes # (issue)

# Checklist:

- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the [documentation](https://github.com/avo-hq/avodocs)
- [ ] I have added tests that prove my fix is effective or that my feature works

## Screenshots & recording

index-violation

## Manual review steps

1. Step 1
1. Step 2

Manual reviewer: please leave a comment with output from the test if that's the case.

### CODE REVIEW

**Review by camilodelvasto — APPROVED** [review]
💯

File: lib/avo/fields/files_field.rb [hunk 1]
```diff
@@ -32,12 +32,9 @@ def to_permitted_param
...
 
-        value.each do |file|
-          # Skip empty values
-          next unless file.present?
-
-          record.send(key).attach file
-        end
+        # attach files in one call to avoid index_active_storage_attachments_uniqueness violation
+        files = value.compact_blank
+        record.send(key).attach(files) unless files.none?
```
Comment (jpawlyn) [inline lib/avo/fields/files_field.rb:37]: On second thoughts, this looks better. Let me just check that it's the same.
```suggestion
        record.send(key).attach(value.compact_blank)
```

**Comment by adrianthedev** [comment]
@camilodelvasto can we add a test to check for the bad behavior that was happening before? Also to check for `nil` values (to see that the `compact_blank`) works.

**Comment by jpawlyn** [comment]
> Looking great! Thank you @jpawlyn for the contribution!
> 
> Good to merge after adding some tests

I haven't been able to re-create the issue in the Avo test suite I'm afraid but I have recreated it in the Active Storage test suite.

Looking at https://github.com/rails/rails/blob/ee661852313ff697d5f3a73a68aab99101ea3ead/activestorage/test/models/attached/many_test.rb#L404-L415 the call to `attach` is taking multiple signed ids. When I change that to 2 separate calls to `attach` ie

**Comment by jpawlyn** [comment]
> > @Paul-Bob I tried writing a test but unfortunately it doesn't fail with the original code. Somehow my app running in production and the activestorage test suite re-creates the issue but not for the Avo test suite.
> 
> Does it fail on Avo local?

Yes it does

**Comment by Paul-Bob** [comment]
> When running my app locally, yes it does fail

Ok, thank you for your effort on this @jpawlyn. I'll have a look
